src/mip/mip_parser.h

```diff
@@ -91,6 +100,24 @@ void* mip_parser_callback_object(const mip_parser* parser);
 
 timestamp_type mip_parser_last_packet_timestamp(const mip_parser* parser);
 
+
+//
+// Diagnostics
+//
+
+#ifdef MIP_ENABLE_DIAGNOSTIC_COUNTERS
+
+uint32_t mip_parser_diagnostic_bytes_read(const mip_parser* parser);
```

Comment:
If we are going to do this based on a compile time definition, I think it might make more sense to use macros here instead of functions so that the same code can compile whether the option is enabled or not

Reply:
What would the macros return if diagnostics are disabled? 0s for everything? I was thinking this case is a bit different from the logging one since the user must explicitly call these functions. Wouldn't one also want the calling code (i.e. user's code) to be #ifdef'd out in that case?